homework-selects/sql_questions.sql

```diff
@@ -3,35 +3,236 @@
 SELECT first_name, last_name, salary, job_title
 FROM employees
          LEFT JOIN jobs USING (job_id);
+         
 -- 2. the first and last name, department, city, and state province for each employee.
+SELECT  first_name, last_name, department_name, city, state_province
+FROM employees 
+    LEFT JOIN departments USING(department_id) 
+    LEFT JOIN locations USING(location_id);
+
 -- 3. the first name, last name, department number and department name, for all employees for departments 80 or 40.
+SELECT  first_name, last_name, department_id, department_name
+FROM employees 
+    LEFT JOIN departments USING(department_id) 
+WHERE department_id IN(80, 40);
+
 -- 4. those employees who contain a letter z to their first name and also display their last name, department, city, and state province.
+SELECT first_name, last_name, department_name, city, state_province
+FROM employees 
+    LEFT JOIN departments USING(department_id) 
+    LEFT JOIN locations USING(location_id)
+    WHERE first_name LIKE '%z%';
+
 -- 5. the first and last name and salary for those employees who earn less than the employee earn whose number is 182.
+SELECT first_name, last_name, salary 
+FROM employees 
+WHERE salary < ( 
+    SELECT salary 
+    FROM employees 
+    WHERE employee_id = 182 
+);
+
 -- 6. the first name of all employees including the first name of their manager.
+SELECT e.first_name AS Employee_first_name, m.first_name AS Manager_first_name
+FROM employees e
+    INNER JOIN employees m ON (m.employee_id = e.manager_id);
+
 -- 7. the first name of all employees and the first name of their manager including those who does not working under any manager.
+SELECT e1.first_name, e2.first_name AS manager_name 
+FROM employees e1 
+    LEFT JOIN employees e2 ON e1.employee_id = e2.manager_id;
+
 -- 8. the details of employees who manage a department.
+SELECT * 
+FROM employees e
+INNER JOIN departments d ON (e.employee_id = d.manager_id);
+
 -- 9. the first name, last name, and department number for those employees who works in the same department as the employee who holds the last name as Taylor.
+SELECT first_name, last_name, department_id 
+FROM employees 
+WHERE department_id IN( 
+    SELECT department_id
+    FROM employees 
+    WHERE last_name = 'Taylor' 
+);
+
 --10. the department name and number of employees in each of the department.
+SELECT department_name, COUNT(employee_id) AS number_of_employees 
+FROM departments 
+    INNER JOIN employees USING(department_id)
+GROUP BY department_name;
+
 --11. the name of the department, average salary and number of employees working in that department who got commission.
+SELECT department_name, AVG(salary) AS average_salary, COUNT(employee_id) AS number_of_employees 
+FROM departments 
+    LEFT JOIN ( 
```

Comment:
I believe left join will get departments even if commission_pct is null, I think we should use inner join

Reply:
and also why not simplify to this :
departments inner join employees using(department_id)
where commision_pct is not null

Reply:
because the query returns the correct information